P8/DataConverters/ShapeletHistogramConverter/FeatureExtraction/src/types/FeatureSet.h

```diff
@@ -0,0 +1,50 @@
+#ifndef FEATUREEXTRACTION_FEATURESET_H
+#define FEATUREEXTRACTION_FEATURESET_H
+
+#include <vector>
+#include "Feature.h"
+#include "SeriesSet.h"
+
+struct FeatureSet : public std::vector<Feature> {
+    inline bool Contains(const Feature &feature) { return std::find(this->begin(), this->end(), feature) != this->end(); }
+    [[nodiscard]] SeriesSet Shapelets() const;
+    [[nodiscard]] std::vector<std::vector<std::string>> FeatureCSV(const std::vector<std::string> &shapeletPaths) const;
+    [[nodiscard]] std::vector<double> GenerateValues(const Series &series) const;
```

Comment:
These method names are maybe a bit too generic. Its quite difficult to see what they do based on the name